src/library_browser.js

```diff
@@ -620,7 +630,11 @@ var LibraryBrowser = {
 
     updateResizeListeners() {
       var canvas = Module['canvas'];
-      Browser.resizeListeners.forEach((listener) => listener(canvas.width, canvas.height));
+      if (Browser.isHiDPIAware) {
+        Browser.resizeListeners.forEach((listener) => listener(canvas.clientWidth, canvas.clientHeight, canvas.width, canvas.height));
+      } else {
+        Browser.resizeListeners.forEach((listener) => listener(canvas.width, canvas.height, canvas.width, canvas.height));
```

Comment:
Can you avoid duplication here by doing something like:

```
var cw =  canvas.width, ch = canvas.height;
if (Browser.isHiDPIAware) {
  cw = canvas.clientWidth;
  hw = canvas.clientHeight;
}
Browser.resizeListeners.forEach((listener) => listener(cw, ch, canvas.width, canvas.height));
```

Reply:
No problem. Changed accordingly